teps/0140-producing-results-in-matrix.md

```diff
@@ -0,0 +1,702 @@
+      taskSpec:
+        params:
+          - name: IMAGE
+          - name: DIGEST
+            default: ""
+        results:
+          - name: IMAGE-DIGEST
+        steps:
+          - name: produce-image-digest
+            image: bash:latest
+            script: |
+              #!/usr/bin/env bash
+              echo "Building image for $(params.IMAGE)"
+              echo -n "$(params.DIGEST)" | sha256sum | tee $(results.IMAGE-DIGEST.path)
+```
+
+
+#### 2. Platforms and Browsers
+
+As a `Pipeline` author, I need to run tests on a combination of platforms and browsers. This will fan out into 9 different `TaskRuns` that will produce a `Result` “report-url” for each combination which can be used in a subsequent `PipelineTask` to fetch the report for each platform-browser combination..
+
+```text
+# platforms
+linux
+windows
+mac
+
+# browsers
+chrome
+firefox
+safari
+```
+
+
+```
+                                                                 clone
+                                                                   |
+                                                                   v
+   --------------------------------------------------------------------------------------------------------------------------
+     |              |              |             |               |                |              |          |          |
+     v              v              v             v               v                v              v          v          v
+linux-chrome  linux-firefox   linux-safari  windows-chrome  windows-firefox  windows-safari  mac-chrome  mac-firefox  mac-safari
+     |              |              |             |               |                |              |          |          |
+     v              v              v             v               v                v              v          v          v
+report-url-0   report-url-1   report-url-2   report-url-3    report-url-4    report-url-5  report-url-6  report-url-7 report-url-8
+```
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: matrix-testing-platform-and-browsers
+spec:
+  tasks:
+    - name: test-platforms-and-browsers
+      matrix:
+        params:
+          - name: platform
+            value:
+              - linux
+              - mac
+              - windows
+          - name: browser
+            value:
+              - chrome
+              - safari
+              - firefox
+      taskSpec:
+        params:
+          - name: platform
+            type: string
+          - name: browser
+            type: string
+        results:
+          - name: report-url
+            type: string
+         steps:
+          - name: produce-report-url
+            image: alpine
+            script: |
+               echo "Running tests on $(params.platform)-$(params.browser)"
+               echo -n "https://api.example/get-report/$(params.platform)-$(params.browser)" | tee $(results.report-url.path)
+```
+
+### API Change
```

Comment:
This sounds like an implementation detail rather than an api change. Is that right? It's an important distinction because we can easily evolve implementation details but need much more consideration for api changes.